BSP/Inc/BSP_UART.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2018-2023 UT Longhorn Racing Solar */
+// BSP UART header file
+
+#ifndef BSP_UART_H
+#define BSP_UART_H
+
+#include <stdint.h>
+#include "stm32f4xx_hal.h" 
+#include "stm32f4xx_hal_rcc.h"
+#include "stm32f4xx_hal_gpio.h"
+#include "stm32f4xx_hal_uart.h"
+#include "stm32f4xx_hal_usart.h"
+
+#define TX_SIZE 60
+#define RX_SIZE 60
+#define QUEUE_SIZE 300
+
+typedef enum {UART_OK, UART_ERROR, UART_BUSY, UART_TIMEOUT} UART_Init_Status;
+typedef enum {WRITE_SUCCESS, WRITE_FAIL} UART_Write_Status;
+
+/**
+ * @brief   Initialize the UART pins as necessary
+ * @param   device the proper UART device to write to defined by the UART_t enum\
+ * @return  Status of init operation
+ */
+UART_Init_Status BSP_UART_Init(USART_HandleTypeDef device);
```

Comment:
All of these return types will need to get updated to BSP_Status from UART_XYZ_Status once standard config is merged